src/platforms/emscripten/src/lib/platform_nifs.c

```diff
@@ -35,16 +47,144 @@ static term nif_atomvm_platform(Context *ctx, int argc, term argv[])
     return EMSCRIPTEN_ATOM;
 }
 
+static void do_run_script(GlobalContext *global, char *script, int sync, int sync_caller_pid)
+{
+    emscripten_run_script(script);
+    if (sync) {
+        Context *target = globalcontext_get_process_lock(global, sync_caller_pid);
+        if (target) {
+            mailbox_send_term_signal(target, TrapAnswerSignal, OK_ATOM);
+            globalcontext_get_process_unlock(global, target);
+        } // else: sender died
+    }
+}
+
+static term nif_emscripten_run_script(Context *ctx, int argc, term argv[])
+{
+    UNUSED(argc);
+
+    int ok;
+    char *str = interop_term_to_string(argv[0], &ok);
+    if (UNLIKELY(!ok || str == NULL)) {
```

Comment:
`interop_term_to_string` should guarantee that when `!ok`, `str` is always `NULL`. Please, let me know if you aware of any exception to this.